src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/content_key_policy.py

```diff
@@ -3,24 +3,196 @@
+                                                       restriction_token_type, issuer, audience)
+
+    # TODO: probably some refactor in the near future to avoid having multiple ANDs.
+    if open_restriction and valid_token_restriction:
+        raise CLIError('You should only use one restriction type.')
 
     if clear_key_configuration:
-        default_configuration = ContentKeyPolicyClearKeyConfiguration()
+        configuration = ContentKeyPolicyClearKeyConfiguration()
 
     if open_restriction:
-        default_restriction = ContentKeyPolicyOpenRestriction()
+        restriction = ContentKeyPolicyOpenRestriction()
+
+    if valid_token_restriction:
+        if _token_restriction_keys_available(symmetric_token_key, rsa_token_key_exponent,
+                                             rsa_token_key_modulus, x509_certificate_token_key) != 1:
+            raise CLIError('You should use alternative (alt) token keys if you have more than one token key.')
+
+        primary_verification_key = None
+        _symmetric_keys = _coalesce_str(alt_symmetric_token_keys).split()
+        _rsa_key_exponents = _coalesce_str(alt_rsa_token_key_exponents).split()
+        _rsa_key_modulus = _coalesce_str(alt_rsa_token_key_modulus).split()
+        _x509_keys = _coalesce_str(alt_x509_certificate_token_keys).split()
+        alternative_keys = []
+        _token_claims = []
+
+        if symmetric_token_key is not None:
+            primary_verification_key = _symmetric_token_key_factory(bytearray(symmetric_token_key, 'utf-8'))
+        elif rsa_token_key_exponent is not None and rsa_token_key_modulus is not None:
+            primary_verification_key = _rsa_token_key_factory(
+                bytearray(rsa_token_key_exponent, 'utf-8'), bytearray(rsa_token_key_modulus, 'utf-8'))
+        elif x509_certificate_token_key is not None:
+            primary_verification_key = _x509_token_key_factory(
+                bytearray(x509_certificate_token_key.replace('\\n', '\n'), 'utf-8'))
+        else:
+            raise CLIError('Invalid token key.')  # This should not happen.. but just in case.
+
+        # Extra validation to make sure exponents and modulus have the same cardinality
+        if len(_rsa_key_exponents) != len(_rsa_key_modulus):
+            raise CLIError('The number of alternative RSA key exponents and modulus differ.')
 
-    options = [ContentKeyPolicyOption(name=policy_option_name,
-                                      configuration=default_configuration,
-                                      restriction=default_restriction)]
+        for key in _symmetric_keys:
+            alternative_keys.append(_symmetric_token_key_factory(key))
+
+        for exp, mod in zip(_rsa_key_exponents, _rsa_key_modulus):
+            alternative_keys.append(_rsa_token_key_factory(exp, mod))
+
+        for key in _x509_keys:
+            alternative_keys.append(_x509_token_key_factory(key))
+
+        for key in token_claims:
```

Comment:
If you don't pass the _token_claims_ parameter, this will try to iterate over a NoneType object and you'll get an error: 

> TypeError: 'NoneType' object is not iterable

Reply:
magnificent mister